### DISCUSSION TITLE
Mixing real image heightmaps with noise?

### DISCUSSION THREAD

**Discussion by Valinakova** [Hoimar/Planet-Generator#29] [Ideas]
The height maps of mountains and canyons taken from space are incredibly realistic. My idea is some way to place those height maps on a curved surface, randomly on any generated planet, and have noise be used to combine it in some way to create original landscapes.  Example of someone using grand canyon height map to combine with a logo. But perhaps it could be used to combine with a noise shape too. 
![image](https://user-images.githubusercontent.com/81381626/134763355-661859c4-a978-472d-b2f0-18aea9c71cac.png)

You can read more here.
[Grand-Canyon-terrain-syntetization-On-the-left-the-user-sketch-the-Grand-Canyon.pdf](https://github.com/Hoimar/Planet-Generator/files/7229370/Grand-Canyon-terrain-syntetization-On-the-left-the-user-sketch-the-Grand-Canyon.pdf)

**Comment by Hoimar** [comment]
That's a great resource, I'll look into that. Coincidentically I've just started to work on this project again.

**Reply by Valinakova** [reply]
Hows the project going so far?

**Reply by Hoimar** [reply]
It's going alright, I'm working on some trickery to allow for larger world scale with single precision floats, like shifting the origin.
Btw, In Godot 4.0 we'll be able to compile with 64bit, so precision won't be an issue anymore.

**Reply by Valinakova** [reply]
Awesome. I've read they should complete double precision at some point in the 4.0 series. So instead of having an accuracy of around a few kilometers, this range will be instead the size of saturns orbit. Will make it easier to create life size planets and stars.  But the origin shifting will still be absolutely crucial, since we want to be able to explore different solar systems or solar systems larger than saturn's orbit without z-fighting. In order for origin shifting to no longer be crucial will probably require something like 128 bit.

**Reply by Hoimar** [reply]
They actually did complete 64bit / double precision support already some months ago.
The Z-Buffer I think is not affected by the floating point precision, that will still require other tweaks, but the logical game world size will be much larger. We can actually work with metres then.

**Reply by Valinakova** [reply]
Well that's good. I was under the impression that double precision will come in 4.1 or 4.2 or some other patched update. Nice to learn it can come out earlier. The double float precision is really going to help with real sized planets and astronomical objects. Godot 4.0 will be the engine that will be simulation friendly while being open source under permissive license. There are a number of game dev influencers who appear to take interest in games of these genre. I will be sure to alert them to the news, as well as to updates on this project.
